src/main/scala/mesosphere/marathon/upgrade/DeploymentActor.scala

```diff
@@ -115,12 +115,15 @@ class DeploymentActor(
     promise.future
   }
 
-  def scaleApp(app: AppDefinition, scaleTo: Int): Future[Unit] = {
+  def scaleApp(app: AppDefinition, scaleTo: Int, toKill: Option[Set[MarathonTask]]): Future[Unit] = {
     val runningTasks = taskTracker.get(app.id)
-    if (scaleTo == runningTasks.size) {
-      Future.successful(())
+    val proposition = ScalingCalculator.scalingProposition(runningTasks, toKill, scaleTo)
```

Comment:
If `ScalingCalculator` was a case class, you could destructure it directly here into tasksToKill/tasksToStart.